node/lnd.py

```diff
@@ -156,17 +157,18 @@ def get_uri(self) -> str:
         return info["uris"][0]
 
     # Check whether the payment has been paid
-    def check_payment(self, rhash: str) -> Tuple[float, float]:
+    def check_payment(self, rhash: str) -> Tuple[Decimal, Decimal]:
         invoice_status = json.loads(
             MessageToJson(self.lnd.lookup_invoice(r_hash_str=b64decode(rhash).hex()))
         )
 
         if "amtPaidSat" not in invoice_status.keys():
-            conf_paid = 0
-            unconf_paid = 0
+            conf_paid = Decimal(0)
+            unconf_paid = Decimal(0)
         else:
             # Store amount paid and convert to BTC units
-            conf_paid = (int(invoice_status["amtPaidSat"]) + 1) / (10 ** 8)
-            unconf_paid = 0
+            conf_paid = Decimal(
+                (int(invoice_status["amtPaidSat"]) + 1) / (10 ** 8))
```

Comment:
I think this `+ 1` was for float errors? But im not certain. Any idea if we drop it now since we are using Decimal? Something like
```
Decimal(amount_sat) / Decimal(10 ** 8)
```